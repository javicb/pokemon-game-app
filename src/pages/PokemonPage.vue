<template>

  <h1 v-if="!pokemon">Cargando...</h1>
  <div v-else>
    <h1>¿Cuál es este pokémon?</h1>
    <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon"/>
    <PokemonOptions :opciones="pokemonArr"/>
  </div>

</template>

<script>
import PokemonOptions from '@/components/PokemonOptions'
import PokemonPicture from '@/components/PokemonPicture'
import getPokemonOptions from '@/helpers/getPokemonOptions'

export default {
  name: 'PokemonPage',
  data() {
    return {
      pokemonArr: [],
      pokemon: null,
      showPokemon: false
    }
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions()
      const random = Math.floor(Math.random() * 4)
      this.pokemon = this.pokemonArr[random]
    }
  },
  components: {
    PokemonOptions,
    PokemonPicture
  },
  mounted () {
    this.mixPokemonArray();
  },
}
</script>

<style>

</style>