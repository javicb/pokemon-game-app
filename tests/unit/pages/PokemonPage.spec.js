import { shallowMount, mount } from '@vue/test-utils';
import PokemonPage from '@/pages/PokemonPage.vue';
import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonPage Component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test('should be match with snapshot', () => {
    // Asert
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call mixPokemonArray when mounting the component', () => {

    // Arrange
    const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray');

    // Act
    mount(PokemonPage);

    // Assert
    expect(mixPokemonArraySpy).toHaveBeenCalled();
  });

  test('It must match the snapshot when the pokemons load', () => {

    // Arrange
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: ''
        }
      }
    });

    // Assert
    expect(wrapper.html()).toMatchSnapshot();
  });

})