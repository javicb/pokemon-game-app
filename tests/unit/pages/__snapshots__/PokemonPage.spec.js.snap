// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component It must match the snapshot when the pokemons load 1`] = `
<div class="fade-in">
  <h1>¿Cuál es este Pokémon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;

exports[`PokemonPage Component should be match with snapshot 1`] = `<h1>Cargando...</h1>`;
